### ISSUE TITLE
Unexpected word in the result-map

### ISSUE THREAD

**Issue by MaxWenzel** [fusion-jena/JaroWinklerSimilarity#1]
I have the following test (used JUnit and AssertJ):

```java
    @Test
    void calculateSimilarity() {
        de.uni_jena.cs.fusion.similarity.jarowinkler.JaroWinklerSimilarity<String> jws =
                de.uni_jena.cs.fusion.similarity.jarowinkler.JaroWinklerSimilarity.with(
                        Arrays.asList("impasse", "vincentd", "vincent","dindy","d","indy"), 0.875d);

        Map<String, Double> result = jws.apply("impasse");

        assertThat(result.keySet()).contains("impasse");
    }
```

I expect the word 'impasse' with a similarity of 1, but I get the word 'dmpasse' (which is not in the input list) with a similarity of 0.904.

**Comment by jmkeil** [comment]
This is a bug in `LinkedNodeTrieSet#addAll(Collection)`:

```java
	@Test
	public void issue1() {
		List<String> words = Arrays.asList("d", "dindy", "impasse");
		LinkedNodeTrieSet root = new LinkedNodeTrieSet(words);
		assertTrue(root.contains("impasse"));
	}
```

Thanks for reporting. I am going to fix that.

**Comment by jmkeil** [comment]
The bug is fixed in 5561ba63d500d0df5d541d7b860986921c466218 and a new version (v1.0.1) has been release and will soon be available on Maven Central.
